From 2cde7a5a33db9ea258f1024880613fa1c4b44705 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 19 Jan 2024 17:58:17 +0100
Subject: [PATCH 0500/1695] plat-stm32mp2: Add all_dtbs build target

Add all_dtbs build target to generate the DTB files from listed DTS
files.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 11b04e139d2b5aa146b9b51cbb433ed312497116 plat-stm32mp2: Add all_dtbs build target]
Change-Id: I8ff3028de10a4861b9264bf23e0bd2add1c4cae9
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/353079
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk |  3 +++
 core/arch/arm/plat-stm32mp2/link.mk | 29 +++++++++++++++++++++++++++++
 2 files changed, 32 insertions(+)
 create mode 100644 core/arch/arm/plat-stm32mp2/link.mk

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index a5810a030..aa0f14ca6 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -4,6 +4,9 @@ flavor_dts_file-257F_EV1 = stm32mp257f-ev1.dts
 flavorlist-MP25 = $(flavor_dts_file-257F_DK) \
 		  $(flavor_dts_file-257F_EV1)
 
+# List of all DTS for this PLATFORM
+ALL_DTS = $(flavorlist-MP25)
+
 # Check if device-tree exist in OP-TEE source code, else search it in external
 # device tree repository
 ifeq ($(wildcard $(arch-dir)/dts/$(CFG_EMBED_DTB_SOURCE_FILE)),)
diff --git a/core/arch/arm/plat-stm32mp2/link.mk b/core/arch/arm/plat-stm32mp2/link.mk
new file mode 100644
index 000000000..df2769fb1
--- /dev/null
+++ b/core/arch/arm/plat-stm32mp2/link.mk
@@ -0,0 +1,29 @@
+include core/arch/arm/kernel/link.mk
+
+#
+# Add target 'all_dtbs' that builds all the dtb for current PLATFORM
+#
+ifneq ($(filter all_dtbs,$(MAKECMDGOALS)),)
+
+.PHONY: all_dtbs
+
+define add-dts-and-build
+# DTS file name in $1
+dts-file := $(if $(wildcard $(arch-dir)/dts/$1), \
+	$(arch-dir)/dts/$1, \
+	$(CFG_EXT_DTS)/$1)
+dtb-file := $(out-dir)/$(arch-dir)/dts/$(1:.dts=.dtb)
+$$(eval $$(call gen-dtb-file,$$(dts-file),$$(dtb-file)))
+
+all_dtbs: $$(dtb-file) FORCE
+
+endef
+
+$(foreach dts, \
+	$(filter-out $(CFG_EMBED_DTB_SOURCE_FILE),$(ALL_DTS)), \
+	$(eval $(call add-dts-and-build,$(dts))) \
+)
+
+all_dtbs: $(out-dir)/$(arch-dir)/dts/$(CFG_EMBED_DTB_SOURCE_FILE:.dts=.dtb)
+
+endif
-- 
2.47.2

